src/components/dialogs/renew.vue

```diff
@@ -0,0 +1,252 @@
+            </v-card-actions>
+          </v-tab-item>
+
+        </v-tabs-items>
+      </v-tabs>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang='ts'>
+import { Component, Mixins, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+import { getFeatureFlag } from '@/plugins'
+import FeeSummary from '@/components/payment/fee-summary.vue'
+import StaffPayment from '@/components/payment/staff-payment.vue'
+import { CreatePaymentParams, FetchFeesParams } from '@/modules/payment/models'
+import { RENEW_MODAL_IS_VISIBLE } from '@/modules/payment/store/types'
+import { FilingTypes } from '@/modules/payment/filing-types'
+import { Jurisdictions, PaymentAction } from '@/enums'
+import { PaymentMixin, PaymentSessionMixin, DisplayedComponentMixin } from '@/mixins'
+import { getBaseUrl } from '@/components/payment/payment-utils'
+import { ActionBindingIF } from '@/interfaces/store-interfaces'
+
+@Component({
+  components: {
+    FeeSummary,
+    StaffPayment
+  }
+})
+export default class RenewDialog extends Mixins(
+  PaymentMixin,
+  PaymentSessionMixin,
+  DisplayedComponentMixin
+) {
+  // the tab indices
+  // NB: these are reversed to reverse the built-in slide transition
+  readonly TAB_STAFF_PAYMENT = 0
+  readonly TAB_RENEW_NAME_REQUEST = 1
+
+  // Refs
+  $refs!: {
+    staffPaymentComponent: StaffPayment
+  }
+
+  // Global getters
+  @Getter getName!: string
+  @Getter isRoleStaff!: boolean
+
+  // Global action
+  @Action toggleRenewModal!: ActionBindingIF
+
+  /** Whether staff payment is valid. */
+  private isStaffPaymentValid = false
+
+  /** The current tab to display. */
+  private currentTab = this.TAB_RENEW_NAME_REQUEST
+
+  /** Whether payment redirection is in progress. */
+  private isLoadingPayment = false
+
+  /** Whether this dialog is visible. */
+  private isVisible = false
+
+  /** Whether this modal should be shown (per store property). */
+  private get showModal (): boolean {
+    return this.$store.getters[RENEW_MODAL_IS_VISIBLE]
+  }
+
+  /** Clears store property to hide this modal. */
+  async hideModal () {
+    this.isLoadingPayment = false
+    await this.toggleRenewModal(false)
+  }
+
+  /** Depending on value, fetches fees and makes this modal visible or hides it. */
+  @Watch('showModal')
+  async onShowModal (val: boolean): Promise<void> {
+    if (val) {
+      // reset tab id
+      this.currentTab = this.TAB_RENEW_NAME_REQUEST
+
+      const params: FetchFeesParams = {
+        filingType: FilingTypes.NM620,
+        jurisdiction: Jurisdictions.BC,
+        priorityRequest: false // no priority on renew
+      }
+
+      // only make visible on success, otherwise hide it
+      if (await this.fetchFees(params)) {
+        this.isVisible = true
+      } else {
+        await this.hideModal()
+      }
+    } else {
+      this.isVisible = false
+    }
+  }
+
+  /** Called when staff clicks "Back" button. */
+  goBack () {
```

Comment:
is this public so it can be called externally?

Reply:
No, I left off `private` because the VS Code Vue tools indicate that the method is not used... it's actually used in the template but something isn't smart enough atm.

I haven't been consistent with tagging internal methods as private because of this. Thoughts?

Reply:
PS I remember reading somewhere that even private methods can be accessed externally. Possibly a TS issue.

Reply:
Interesting! I will have a look my editor as well.
Either way, no action required, just curious of the approach :)

Reply:
Ideall,y we would mark all local methods as private, but I don't like seeing "blah is declared but its value is never read" (with the suggested fix to "remove unused declarations")!

Any thoughts on what we should do (for code style / consistency)?

Reply:
Good question, i'm not sure off the top of my head.
 If it's a linting issue specific to vue/vue templates perhaps it's something we could configure?
 
It seems the Class and Template are 2 separate things, so declaring private with no class usage seems like the error is justified. 
Maybe declaring without private is the way!
Not sure but either way i'm ok with this, as is :)

Reply:
Yeah, same things I struggled with.

PS I believe it's [Vetur](https://vuejs.github.io/vetur/) / IntelliSense that is complaining about the private methods not being used. It doesn't look like it's configurable.

PPS Note that, in other UI frameworks, the template and script are indeed separate files. But I kinda think they're part of the same "class", fwiw.

Reply:
Interesting! Yea, i'll need to look a little closer at how components are bundled. I'm oblivious!